cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -469,44 +468,92 @@ def create_config(server, creds):
+
+    extra_args = parse_extra_args(extra_args_key)
+
+    args = {}
+    for arg in prev_args:
+        # remove previous args by setting to null
+        args[arg] = 'null'
+    for k, v in base_args.items():
+        args[k] = v
+    for k, v in extra_args.items():
+        args[k] = v
+
+    cmd = ['snap', 'set', service] + ['%s=%s' % item for item in args.items()]
+    check_call(cmd)
+
+    db.set(prev_args_key, args)
+
+
+def configure_kubelet(dns):
     layer_options = layer.options('tls-client')
     ca_cert_path = layer_options.get('ca_certificate_path')
     server_cert_path = layer_options.get('server_certificate_path')
     server_key_path = layer_options.get('server_key_path')
 
-    kubelet_opts = FlagManager('kubelet')
-    kubelet_opts.add('require-kubeconfig', 'true')
-    kubelet_opts.add('kubeconfig', kubeconfig_path)
-    kubelet_opts.add('network-plugin', 'cni')
-    kubelet_opts.add('v', '0')
-    kubelet_opts.add('address', '0.0.0.0')
-    kubelet_opts.add('port', '10250')
-    kubelet_opts.add('cluster-dns', dns['sdn-ip'])
-    kubelet_opts.add('cluster-domain', dns['domain'])
-    kubelet_opts.add('anonymous-auth', 'false')
-    kubelet_opts.add('client-ca-file', ca_cert_path)
-    kubelet_opts.add('tls-cert-file', server_cert_path)
-    kubelet_opts.add('tls-private-key-file', server_key_path)
-    kubelet_opts.add('logtostderr', 'true')
-    kubelet_opts.add('fail-swap-on', 'false')
-
-    kube_proxy_opts = FlagManager('kube-proxy')
-    kube_proxy_opts.add('cluster-cidr', cluster_cidr)
-    kube_proxy_opts.add('kubeconfig', kubeproxyconfig_path)
-    kube_proxy_opts.add('logtostderr', 'true')
-    kube_proxy_opts.add('v', '0')
-    kube_proxy_opts.add('master', random.choice(api_servers), strict=True)
+    kubelet_opts = {}
+    kubelet_opts['require-kubeconfig'] = 'true'
+    kubelet_opts['kubeconfig'] = kubeconfig_path
+    kubelet_opts['network-plugin'] = 'cni'
+    kubelet_opts['v'] = '0'
+    kubelet_opts['address'] = '0.0.0.0'
+    kubelet_opts['port'] = '10250'
+    kubelet_opts['cluster-dns'] = dns['sdn-ip']
+    kubelet_opts['cluster-domain'] = dns['domain']
+    kubelet_opts['anonymous-auth'] = 'false'
+    kubelet_opts['client-ca-file'] = ca_cert_path
+    kubelet_opts['tls-cert-file'] = server_cert_path
+    kubelet_opts['tls-private-key-file'] = server_key_path
+    kubelet_opts['logtostderr'] = 'true'
+    kubelet_opts['fail-swap-on'] = 'false'
+
+    privileged = is_state('kubernetes-worker.privileged')
+    kubelet_opts['allow-privileged'] = 'true' if privileged else 'false'
+
+    if is_state('kubernetes-worker.gpu.enabled'):
+        if get_version('kubelet') < (1, 6):
+            hookenv.log('Adding --experimental-nvidia-gpus=1 to kubelet')
+            kubelet_opts['experimental-nvidia-gpus'] = '1'
+        else:
+            hookenv.log('Adding --feature-gates=Accelerators=true to kubelet')
+            kubelet_opts['feature-gates'] = 'Accelerators=true'
```

Comment:
I need to test that the GPU functionality still works with this change. I don't think our automated tests cover it.

Reply:
Tested with a GPU worker. This worked how I would expect, so we should be good.